arevyn24-assignment-4/src/arevyn24/assignment/pkg4/Assignment4Methods.java

```diff
@@ -0,0 +1,329 @@
+            "very distant lands", "With Jake the Dog", "And Finn the Human",
+            "The fun will never end,", "It's Adventure Time!"
+        };    
+
+        //title and description
+        print("SONG LYRICS LOOP\n" + "\tEnter a number from"
+                + " 1-8. This will be the number of times the loop"
+                + " executes: ");
+        numLoops = keyboard.nextInt();
+
+        do { 
+            /*If input is not a number between 1-8, prompt the user
+              to enter a new value. Keep looping until they enter a
+              correct value*/
+            while (numLoops > 8 || numLoops <= 0) {
+                print("Oops! Try again. Enter a number from"
+                                    + " 1-8.\n");
+                numLoops = keyboard.nextInt();
+            }
+            /*
+              In this for loop, the variable i represents the location
+              inside the lyrics[] array. numLoops is the user-chosen
+              number that decides how many lines of lyrics will be 
+              printed Each time the loop runs, i is increased, making
+              the next line of the song print out
+            */
+            for(int i=0; i < numLoops; i++) {
+                print("\t\t" + lyrics[i] +"\n");
+            } 
+            print("\nEnter a number 1-8 to loop again or 9 to"
+                    + " return to main menu: ");
+            numLoops = keyboard.nextInt();
+        } while (numLoops != 9);
+    };      //end loopingLyrics method
+    
+    
+    /**
+    * slotMachine lets the user play a text version of a slot machine game. Enter
+    * a bet and double or triple the bet amount based on the number of matching
+    * symbols
+    * @param keyboard Scanner object reads user input
+    */  
+    public static void slotMachine(Scanner keyboard) {
+      //Chapter 4 Exercise 22: Slot Machine Simulation (page 267)
+      //Objective: Create a program that simulates a slot machine
+        Integer betAmt, winnings;
+        Random spin = new Random(); //random object generates new random number
+        Integer[] randoms = {0,0,0}; //will hold values to be converted to strings
+        String[] slotValues = {" "," "," "};
+        String slot1 = " ",slot2 = " ", slot3 = " " ; //hold text representation of randoms
+
+      //read user bet amount
+        print("SLOT MACHINE\n"
+                + "\tPlace Your Bet (dollar amount [Integer]): ");
+        betAmt = keyboard.nextInt();
+
+      //validates user input betAmt
+        while(betAmt < 1 || betAmt > 100) {
+            print("\tTry an amount between 1 and 100");
+            betAmt = keyboard.nextInt();  
+        }
+
+      /*assign random values to each of the three slots
+       *spin.nextInt(6) will generate a random number from 0 and 5
+       * slotValue() method assigns text value to each slot
+       */
+        for(int i = 0; i < randoms.length; i++) {
+            randoms[i] = spin.nextInt(6);
```

Comment:
Pseudo random numbers are hard. `.nextInt(6)` in this context might return the same number every time.

You have two choices:

1. Create a new instance of Random for each loop `randoms[i] = (new Random()).nextInt(6);
2. get a random window of the seedfile and you can skip the loop assignment for randoms `randoms = spin.ints(1,7).limit(3).toArray();`

That second one is a little heavier. I think randoms are fun personally so maybe we will spend a little more time on it in class later this semester.